Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.DynamicExpression.cs

```diff
@@ -49,34 +51,39 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public DynamicExpression Update (IEnumerable<Expression> arguments)
     {
+      Contract.Ensures(Contract.Result<DynamicExpression>() != null);
```

Comment:
@sharwell I don't think we can and should add this precondition. You may find this code: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/DynamicExpression.cs,710

This code means that `null` is an absolutely valid input.